src/pudl/transform/ferc1.py

```diff
@@ -311,6 +362,60 @@ def transform_end(self, df: pd.DataFrame) -> pd.DataFrame:
         """Enforce the database schema and remove any cached dataframes."""
         return self.enforce_schema(df)
 
+    def normalize_metadata_xbrl(
+        self, xbrl_fact_names: list[str] | None
+    ) -> pd.DataFrame:
+        """Normalize XBRL metadata, select table-specific rows, and transform them.
+
+        In order to select the relevant rows from the normalized metadata, this function
+        needs to know which XBRL facts pertain to the table being transformed. These are
+        the names of the data columns in the raw XBRL data. To avoid creating a separate
+        dependency on the FERC 1 XBRL DB, we defer the assignment of this class
+        attribute until :meth:`Ferc1AbstractTableTransformer.process_xbrl` is called,
+        and read the column labels from the input dataframes.
+        """
+        # If the table has no XBRL metadata, return immediately:
+        if not self.xbrl_metadata_json:
+            return pd.DataFrame()
+
+        normed_meta = (
+            pd.json_normalize(self.xbrl_metadata_json)
```

Comment:
if we add the table name into the json as a top-layer of nesting we can avoid:
* passing the entire json into each table transformer class
* we don't need to get the list of xbrl column names from `process_xbrl`

Reply:
this is a later suggestion

Reply:
I agree, but I think we'll have to come back to this and ask @zschira to modify how the taxonomy metadata is getting extracted.